packages/realm_generator/lib/src/realm_model_info.dart

```diff
@@ -118,12 +103,47 @@ class RealmModelInfo {
       yield '$name freeze() => RealmObjectBase.freezeObject<$name>(this);';
       yield '';
 
+      // Encode
+      yield 'static EJsonValue _encode$name($name value) {';
```

Comment:
The generated code here is quite a bit different from the generated code for ejson annotated objects. Is this intentional?

Reply:
Yes, it is intentional. The `ejson_generator` cannot change the type it generates codecs for. Hence the encoder and decoder are free standing functions. Also, they need to be public, so that the user can register them - there is no good place to register them implicitly.

Now, that realm_generator handles the generation for realm models, they can be internal to the class, they can be registered together with the factory (ie. when opening the schema), and hence can be private to the generated package (file).